nethttp/server.go

```diff
@@ -62,7 +62,31 @@ func MWComponentName(componentName string) MWOption {
 //	        return "HTTP " + r.Method + ":/api/customers"
 //      }),
 //   )
+//
+// MiddlewareWithSpan could add additional tag to the span.
+// The options allow fine tuning the behavior of the middleware.
+//
+// Example:
+//   mw := nethttp.MiddlewareWithSpan(
+// 		tm.tracer,
+// 		handler,
+// 		[]nethttp.AdditionalSpan{
+// 			func(r *http.Request) (string, string) {
+// 				return "http.URI", r.URL.EscapedPath()
+// 			},
+// 		},
+// 		nethttp.OperationNameFunc(func(r *http.Request) string {
+// 			return "HTTP " + r.Method + " " + pattern
+// 		}))
+
 func Middleware(tr opentracing.Tracer, h http.Handler, options ...MWOption) http.Handler {
+	var noneAdditionnalTag []AdditionalTag
+	return MiddlewareWithTag(tr, h, noneAdditionnalTag, options...)
+}
+
+type AdditionalTag func(r *http.Request) (string, string)
+
+func MiddlewareWithTag(tr opentracing.Tracer, h http.Handler, additionalTag []AdditionalTag, options ...MWOption) http.Handler {
```

Comment:
the main Middleware method is already using an extensible functional options pattern. You can either add an option for tags, or an option for AfterSpanStarted function that can do anything.

Reply:
Is this ok?  Just take request and span as input, do anything in the function option.   

Or taking MWOption type of parameter to set tag to the span is better?